pymeasure/display/widgets/dock_widget.py

```diff
@@ -84,11 +99,25 @@ def _setup_ui(self):
             self.docks.append(dock)
 
     def _layout(self):
+        hbox = QtWidgets.QHBoxLayout()
+        hbox.addStretch(5)
+        hbox.addWidget(self.save_layout_button, 1)
```

Comment:
While I (at this moment) also don't really see another option, I do have to admit that I find the single `save` button at the top right a bit uggly / a waste of space (which I think is rather precious when using the DockWidget)

Reply:
I agree and I figured this PR would prompt a discussion where widget buttons should go in general. Should widget buttons be confined to their QWidget or should there be a standard place in the GUI for them?

If we assume widgets with buttons will be implemented in a `ManagedWindowBase` implementation, like DockWindow, then we may be able to put widget buttons in a standard place.

GUI real-estate-wise, there is space around  the `BrowserWidget` besides "Open", or the base ManagedWindowBase QWidget around "Queue" and "Abort".

Reply:
I'm a bit reluctant to place the buttons at another place than the widget itself; I foresee that it creates confusion as to what button does what and relates to which part of the interface (plus it'd need some logic to switch between buttons if you switch tabs). Hence I'd keep buttons (somehow) within the widget itself.

In #771, the actions are put in context menu (i.e. right-click somewhere), so I was thinking maybe that would be an idea, to put the save action (and maybe then also a load?) in a context menu; either in the widget itself, but seeing that that is a pyqtgraph widget, I'm not sure if that is the easiest option, so an alternative would be a context menu of the tab-bar. What do you think of this idea?

Reply:
I tried a couple things with the context menu. If we setup a context menu for the overall `DockWidget`, it will override the `PlotWidget.plot_frame` context menu aka `pyqtgraph.PlotWidget` context menu. Each of those context menus are specific to the plot, for instance checking the "Plot Options -> Transforms -> Log X" option. 

I pushed a change to add "Save Dock Layout" to the context menu of each `pyqtgraph.PlotWidget`, and that works, you can pull down the branch again to test that out.

However the context menu is only accessible in these GUI regions, since `DockWidget` creates a number of `PlotWidgets`:
![image](https://user-images.githubusercontent.com/5920433/210451807-747f3649-faf2-4af1-b704-f615861cbe92.png)

I'll see if there is a way for the widget (red regions) to have a context menu without overriding the `pyqtgraph.PlotWidget` context menu.

Reply:
Well, that was easier than I thought. There is a helpful Qt function `widget.childAt(position)` which can be used to only create a context menu at positions in `DockWidget` that aren't a `pyqtgraph.PlotWidget`.

Try out the latest commit and let me know if that is more intuitive for users. I'll work on documentation.

Reply:
I'm still not sure if the context menu is the best place for this useful functionality. It should be more obvious then a hidden context menu, but I'm not sure of other options. The tutorial and documentation can demonstrate the context menu.

The context menu will work to set a precedent for not allowing button mania in pymeasure UI design.

Reply:
Hmm, I played around with it, and it seems to work nicely.
I do understand your concern though (re. the fact that it is less obvious than a button), I'm incline to lean towards the context menu nevertheless (over the button), especially as I would expect it is not a button you'll use every time (or rather not even very often) you use the software, but rather use it one of the first few times you setup your experiment and leave it (more or less) alone for the remainder of the use. Again, I can be wrong, but even for many uses, I would not be annoyed if the action is located in the context menu.

Reply:
Regarding the "expensive real estate" around the widgets, I made a PR (#822) to move the coordinates label, to save some space around the plots

Reply:
Yea the context menu is growing on me. I'll check out that PR

Reply:
Let's leave it as it is then.